metabase/metabase
# (2/n) rename `DashboardHeader.tsx` to `DashboardHeaderView` to avoid collision with `DashboardHeader.jsx`

Part of epic https://github.com/metabase/metabase/issues/29502

### Description

Just a small refactor to improve component names a bit.

### How to verify

Open a dashboard and confirm it and its header are still working.


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/29684)
